compiler/src/iree/compiler/Codegen/Utils/Utils.cpp

```diff
@@ -1130,10 +1132,159 @@ int64_t getMinElementBitwidth(linalg::LinalgOp linalgOp) {
   return bitwidth;
 };
 
+//===---------------------------------------------------------------------===//
+// Bufferization utility functions
+//===---------------------------------------------------------------------===//
+
+/// Get strides for row-major oredering of a tensor with the given `shape`.
+static SmallVector<int64_t> getStridesFromShape(ArrayRef<int64_t> shape) {
+  if (shape.empty()) {
+    return {};
+  }
+  SmallVector<int64_t> strides(shape.size(), ShapedType::kDynamic);
+  strides.back() = 1;
+  for (int i = strides.size() - 1; i > 0; --i) {
+    if (ShapedType::isDynamic(shape[i])) {
+      break;
+    }
+    strides[i - 1] = strides[i] * shape[i];
+  }
+  return strides;
+}
+
+static MemRefType
+getMemrefTypeForTensor(IREE::TensorExt::DispatchTensorType tensorType,
+                       MemRefLayoutAttrInterface layout = {},
+                       Attribute memorySpace = {}) {
+  return MemRefType::get(tensorType.getShape(),
+                         tensorType.getBoundElementType(), layout, memorySpace);
+}
+
+Value findOrCreateSubspanBuffer(
+    RewriterBase &rewriter, IREE::HAL::InterfaceBindingSubspanOp subspanOp) {
+  // Ensure that this a tensor subspan op.
+  auto shapedType = llvm::dyn_cast<IREE::TensorExt::DispatchTensorType>(
+      subspanOp.getResult().getType());
+  assert(shapedType && shapedType.hasRank());
```

Comment:
I think it is better to encode the comment into the error message. Otherwise, people need to get back to the code and figure the reason. And the comment is the reason. (There is no direct connection between the variable name and the type.)

```suggestion
  assert((shapedType && shapedType.hasRank()) && "expected the result of subspanOp is DispatchTensorType");
```

(We've seen bugs like assertion(A || B && "error message"), so it is better to always wrap the conditions with braces if there are multiple checks.)